### PR TITLE
update error spec

### PR DESCRIPTION
**Pull request by Binyang2014**

### CODE REVIEW

File: src/k8s-job-exit-spec/config/k8s-job-exit-spec.yaml [hunk 1]
```diff
@@ -748,15 +772,16 @@ spec:
   phrase: ContainerSigSegvReceived
   issuer: USER_CONTAINER
   causer: USER_CONTAINER
-  type: USER_FAILURE
+  type: UNKNOWN_FAILURE
   stage: RUNNING
   behavior: PERMANENT
```
Comment (yqwang-ms) [inline src/k8s-job-exit-spec/config/k8s-job-exit-spec.yaml:777]: >PERMANENT [](start = 12, length = 9)

UNKNOWN,


refer:

- code: -300
  phrase: PodFailedWithoutFailedContainer
  issuer: PAI_FC
  causer: UNKNOWN
  type: UNKNOWN_FAILURE
  stage: LAUNCHING
  behavior: UNKNOWN
  reaction: RETRY_TO_MAX
  reason: "Pod failed but without any failed container, may be stopped by K8S"
  repro:
  - "Submit a job which will be rejected by kubelet"
  solution:
  - "Check diagnostics and find root cause"
  - "Wait result from next retry"
  - "Contact Cluster Admin"

File: src/k8s-job-exit-spec/config/k8s-job-exit-spec.yaml [hunk 2]
```diff
@@ -614,6 +614,30 @@ spec:
...
+# Description: User Container issued failures:
+# -> Involuntary failures caused by hardware
+###########################################################################
+- code: 128
+  phrase: ContainerMayFailDueToGpuDeviceEccError
+  issuer: USER_CONTAINER
+  causer: PAI_HW
+  type: PLATFORM_FAILURE
+  stage: RUNNING
+  behavior: TRANSIENT
```
Comment (yqwang-ms) [inline src/k8s-job-exit-spec/config/k8s-job-exit-spec.yaml:629]: >TRANSIENT [](start = 12, length = 9)

behavior: UNKNOWN

File: src/k8s-job-exit-spec/config/k8s-job-exit-spec.yaml (hunk 1 above)
Comment (Binyang2014) [reply src/k8s-job-exit-spec/config/k8s-job-exit-spec.yaml]: Fixed

File: src/k8s-job-exit-spec/config/k8s-job-exit-spec.yaml (hunk 2 above)
Comment (Binyang2014) [reply src/k8s-job-exit-spec/config/k8s-job-exit-spec.yaml]: Fixed

File: src/k8s-job-exit-spec/config/k8s-job-exit-spec.yaml [hunk 3]
```diff
@@ -748,15 +772,16 @@ spec:
   phrase: ContainerSigSegvReceived
   issuer: USER_CONTAINER
   causer: USER_CONTAINER
```
Comment (yqwang-ms) [inline src/k8s-job-exit-spec/config/k8s-job-exit-spec.yaml:774]: >USER_CONTAINER [](start = 10, length = 14)

causer -> UNKNOWN,

**Pull request merged**